nova/api/openstack/compute/servers.py

```diff
@@ -1108,6 +1108,10 @@ def delete(self, req, id):
         except exception.InstanceInvalidState as state_error:
             common.raise_http_conflict_for_instance_invalid_state(state_error,
                     'delete', id)
+        except exception.ComputeServiceUnavailable:
+            msg = _("Compute service is not available at the moment.\
+                    Please try again later.")
+            raise webob.exc.HTTPInternalServerError(explanation=msg)
```

Comment:
I think we should raise a 503 instead of a 500. "Service unavailabe" sounds better than the generic "Internal Server Error". Are we able to specify any headers here? https://www.webfx.com/web-development/glossary/http-status-codes/what-is-a-503-status-code/ states that clients could support a "Retry-After" header.